marlowe/security.md

```diff
@@ -0,0 +1,112 @@
+$ cardano-cli address info --address addr1w94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvq8evxaf | jq -r .base16
+716a9391d6aa51af28dd876ebb5565b69d1e83e5ac7861506bd29b56b0
+
+$ cardano-cli address info --address addr1z94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvxr7cz4mu6gh005gdck67p7y9d8s8zsfgjkcdy75mrjh6jqe7ks37 | jq -r .base16
+116a9391d6aa51af28dd876ebb5565b69d1e83e5ac7861506bd29b56b0c3f6055df348bbdf443716d783e215a781c504a256c349ea6c72bea4
+
+$ cardano-cli address info --address addr1z94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvxn564zluen7mc4t7lcw33v2gz5jlkcdyqh9628mq5u9m3qfuhp8s | jq -r .base16
+116a9391d6aa51af28dd876ebb5565b69d1e83e5ac7861506bd29b56b0d3a6aa2ff333f6f155fbf87462c5205497ed8690172e947d829c2ee2
+```
+
+#### What is the current contract and its state?
+
+The pre-transaction state of the contract is defined in the Plutus `Datum` associated with the UTxO being spent from the Marlowe script address, and this datum must be provided in the transaction.  This datum contains . . .
+- the balances of the accounts internal to the contract,
+- the history of choices previously made up to this point in the contract's execution,
+- the current values of the contract's bound variables, and
+- the part of the contract that remains to be executed.
+
+The `Datum` can be extracted from the unsigned transaction body and deserialized to [`Language.Marlowe.Core.V1.Semantics.MarloweData`](src/Language/Marlowe/Core/V1/Semantics.hs) using the function `Plutus.V2.Ledger.Api.fromData`. Alternatively, the command-line tool `marlowe log --show contract` will display the on-chain history of the contract.
+
+
+#### What input is being applied to the contract?
+
+The input being applied to the contract in the transaction is defined in the Plutus `Redeemer` associated with spending the UTxO from the Marlowe script address, along with the slot validity interval for the transaction specified in the transaction body. The input is a sequence of zero or more deposits, choices, and notifications. The consequences of applying this input to the contract can be studied using a tool like Marlowe Playground or `marlowe-cli run prepare`.
+
+The `Redeemer` can be extracted from the unsigned transaction body and deserialized to [`Language.Marlowe.Scripts.MarloweInput`](src/Language/Marlowe/Scripts.hs) using the function `Plutus.V2.Ledger.Api.fromData`. The command-line tool `marlowe-cli util slotting` will compute the relationship between the slots mentioned in the validity interval to the POSIX times in the contract.
+
+
+#### What else is occurring in the transaction?
+
+The unsigned transaction may contain other spending and payments beyond that specified for the Marlowe contract. This can be examined with the tool `cardano-cli transaction view`.
+
+
+### Marlowe Payout Validator
+
+In addition to the *semantics validator* discussed above, Marlowe has a *payout validator*. The same security consideration hold for it. Its script hash can be computed as follows:
```

Comment:
Suggest adding an "s" to "consideration". --> 

"The same security considerations hold for it."